tests/test_api.py

```diff
@@ -55,3 +56,49 @@ def test_passthrough(self):
         """Test instances are passed through unmodified."""
         a = A(name='charlie')
         self.assertEqual(a, self.resolver.make(a))
+
+
+def _dummy_training_function(config):
+    """A dummy training function without actual training."""
+    # instantiate from configuration
+    to_resolve = config["choice"]
+    if isinstance(to_resolve, dict):
+        query = to_resolve.pop("query")
+        kwargs = to_resolve
+    else:
+        query = to_resolve
+        kwargs = None
+    resolver = Resolver([A, B, C], base=Base)
+    instance = resolver.make(query, pos_kwargs=kwargs)
+    if instance.name == "charlie":
+        mean_loss = 1.0
+    else:
+        mean_loss = 2.0
+
+    import ray.tune
```

Comment:
I added `ray` as a testing extra, so you can assume that it's importable in tests